geomesa-blobstore/geomesa-blobstore-core/src/main/scala/org/locationtech/geomesa/blob/core/AccumuloBlobStore.scala

```diff
@@ -78,6 +79,32 @@ class AccumuloBlobStore(ds: AccumuloDataStore) {
     }
   }
 
+  def delete(id: String): Unit = {
+    // TODO: Get Authorizations using AuthorizationsProvider interface
+    // https://geomesa.atlassian.net/browse/GEOMESA-986
+    val bd = connector.createBatchDeleter(blobTableName, new Authorizations(), bwc.getMaxWriteThreads, bwc)
+    bd.setRanges(List(new Range(new Text(id))))
+    bd.delete()
+    bd.close()
+    deleteFeature(id)
+  }
+
+  private def deleteFeature(id: String): Unit = {
+    val removalFilter = Filters.ff.id(Filters.ff.featureId(id))
+    val fd = ds.getFeatureWriter(blobFeatureTypeName, removalFilter, Transaction.AUTO_COMMIT)
+    try {
+      while (fd.hasNext) {
+        fd.next()
+        fd.remove()
```

Comment:
unless the feature source needs to be cast to something else,  both SimpleFeatureSource and FeatureSource don't implement `removeFeatures` at least according to the ide.